src/uct/ib/dc/dc_mlx5.c

```diff
@@ -1218,6 +1218,24 @@ static uct_rc_iface_ops_t uct_dc_mlx5_iface_ops = {
     .fc_handler = uct_dc_mlx5_iface_fc_handler,
 };
 
+static void
+uct_dc_mlx5_ep_pending_purge_and_progress(uct_ep_h tl_ep,
+                                          uct_pending_purge_callback_t cb,
+                                          void *arg)
+{
+    uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_ep->iface,
+                                                uct_dc_mlx5_iface_t);
+    uct_dc_mlx5_ep_t *ep       = ucs_derived_of(tl_ep, uct_dc_mlx5_ep_t);
+    uint8_t dci                = ep->dci;
+    uint8_t pool_idx;
+
+    uct_dc_mlx5_ep_pending_purge(tl_ep, cb, arg);
+    if (dci != UCT_DC_MLX5_EP_NO_DCI) {
+        pool_idx = uct_dc_mlx5_iface_dci_pool_index(iface, dci);
+        uct_dc_mlx5_iface_progress_pending(iface, pool_idx);
```

Comment:
it's not ok to progress pendiong from purge
should happened only from progress context
is this fix critical?

Reply:
yes, pending_purge without progress cause arbiter ordering failure
I can add deferred DCI releasing, but we have to make deep refactoring of uct_dc_mlx5_ep_pending_purge routine to not free DCI

Reply:
i think just remove this line:
```
    if (ep->dci != UCT_DC_MLX5_EP_NO_DCI) {
        uct_dc_mlx5_iface_dci_free(iface, ep);
    }
```
and schedule progress function which goes over all allocated dcis, and releases it if it belongs to and ep w/o outstanding and w/o pending